device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt

```diff
@@ -449,9 +455,76 @@ class CalendarDelegate : PluginRegistry.RequestPermissionsResultListener {
+            else {
+                if (!followingInstances!!) { // Only this instance
+                    val exceptionUriWithId = ContentUris.withAppendedId(Events.CONTENT_EXCEPTION_URI, eventIdNumber)
+                    val values = ContentValues()
+                    val instanceCursor = CalendarContract.Instances.query(contentResolver, EVENT_INSTANCE_DELETION, startDate!!, endDate!!)
+
+                    for (x in 0 until instanceCursor.count) {
+                        instanceCursor.moveToNext()
+                        val foundEventID = instanceCursor.getLong(EVENT_INSTANCE_DELETION_ID_INDEX)
+
+                        if (eventIdNumber == foundEventID) {
+                            values.put(Events.ORIGINAL_INSTANCE_TIME, instanceCursor.getLong(EVENT_INSTANCE_DELETION_BEGIN_INDEX))
+                            values.put(Events.STATUS, Events.STATUS_CANCELED)
+                        }
+                    }
+
+                    val deleteSucceeded = contentResolver?.insert(exceptionUriWithId, values)
+                    finishWithSuccess(deleteSucceeded != null, pendingChannelResult)
+                }
+                else { // This and following instances
+                    val eventsUriWithId = ContentUris.withAppendedId(Events.CONTENT_URI, eventIdNumber)
+                    val values = ContentValues()
+                    val instanceCursor = CalendarContract.Instances.query(contentResolver, EVENT_INSTANCE_DELETION, startDate!!, endDate!!)
+
+                    for (x in 0 until instanceCursor.count) {
+                        instanceCursor.moveToNext()
+                        val foundEventID = instanceCursor.getLong(EVENT_INSTANCE_DELETION_ID_INDEX)
+
+                        if (eventIdNumber == foundEventID) {
+                            val newRule = org.dmfs.rfc5545.recur.RecurrenceRule(instanceCursor.getString(EVENT_INSTANCE_DELETION_RRULE_INDEX))
+                            val lastDate = instanceCursor.getLong(EVENT_INSTANCE_DELETION_LAST_DATE_INDEX)
+
+                            if (lastDate > 0 && newRule.count != null && newRule.count > 0) { // Update occurrence rule
+                                val cursor = CalendarContract.Instances.query(contentResolver, EVENT_INSTANCE_DELETION, startDate, lastDate)
+                                for (y in 0 until cursor.count) {
+                                    cursor.moveToNext()
+                                    if (eventIdNumber == cursor.getLong(EVENT_INSTANCE_DELETION_ID_INDEX)) {
+                                        newRule.count--
+                                    }
+                                }
+                                cursor.close()
+                            }
+                            else { // Indefinite and specified date rule
+                                val cursor = CalendarContract.Instances.query(contentResolver, EVENT_INSTANCE_DELETION, 946645200000, startDate - 1) // 946645200000 = 01/01/2000
```

Comment:
change constant - to long max

Reply:
Updated (using constants: `startDate - DateUtils.YEAR_IN_MILLIS`)